projects/observability/src/shared/dashboard/widgets/top-n/data/top-n-data-source.model.ts

```diff
@@ -1,41 +1,29 @@
-import { EnumPropertyTypeInstance, ENUM_TYPE } from '@hypertrace/dashboards';
-import {
-  AttributeSpecificationModel,
-  GraphQlDataSourceModel,
-  MetricAggregation
-} from '@hypertrace/distributed-tracing';
-import { Model, ModelProperty, ModelPropertyType, NUMBER_PROPERTY } from '@hypertrace/hyperdash';
+import { GraphQlDataSourceModel } from '@hypertrace/distributed-tracing';
+import { Model, ModelProperty, NUMBER_PROPERTY, STRING_PROPERTY } from '@hypertrace/hyperdash';
 import { Observable, of } from 'rxjs';
 import { map } from 'rxjs/operators';
+import { ExploreRequestContext } from '../../../../components/explore-query-editor/explore-visualization-builder';
 import { Entity, entityIdKey, entityTypeKey, ObservabilityEntityType } from '../../../../graphql/model/schema/entity';
-import { MetricAggregationSpecification } from '../../../../graphql/model/schema/specifications/metric-aggregation-specification';
-import { EntitiesResponse } from '../../../../graphql/request/handlers/entities/query/entities-graphql-query-builder.service';
+import { ExploreSpecification } from '../../../../graphql/model/schema/specifications/explore-specification';
+import { ExploreSpecificationBuilder } from '../../../../graphql/request/builders/specification/explore/explore-specification-builder';
 import {
-  EntitiesGraphQlQueryHandlerService,
-  ENTITIES_GQL_REQUEST
-} from '../../../../graphql/request/handlers/entities/query/entities-graphql-query-handler.service';
+  ExploreGraphQlQueryHandlerService,
+  ExploreQueryContextType,
+  EXPLORE_GQL_REQUEST,
+  GraphQlExploreResponse
+} from '../../../../graphql/request/handlers/explore/explore-graphql-query-handler.service';
 
 @Model({
   type: 'top-n-data-source'
 })
 export class TopNDataSourceModel extends GraphQlDataSourceModel<TopNWidgetDataFetcher> {
   @ModelProperty({
-    key: 'entity',
+    key: 'context',
```

Comment:
Is there any benefit of keeping the entity based data source? If yes, I think we should create an explore-top-n-data-source instead?

Reply:
I don't know if there's any benefit to it. All the existing usages have to be moved over to an explore query to support some of the upcoming requirements.

We can always bring it back if we want I guess. No strong preference there.